test/hydrated_bloc_test.dart

```diff
@@ -84,114 +103,485 @@ class MyMultiHydratedBloc extends HydratedBloc<int, int> {
+
+          expect(hydratedStorage.read('CounterBloc'), '{"value":4}');
+
+          await hydratedStorage.delete('CounterBloc');
+          expect(hydratedStorage.read('CounterBloc'), isNull);
+        });
+      });
     });
 
-    group('clear', () {
-      test('calls delete on storage', () async {
-        await bloc.clear();
-        verify(storage.delete('MyHydratedBloc')).called(1);
+    group('Custom Storage Directory', () {
+      HydratedBlocStorage hydratedStorage;
+
+      tearDown(() async {
+        await hydratedStorage.clear();
+      });
+
+      group('read', () {
+        test('returns null when file does not exist', () async {
+          hydratedStorage = await HydratedBlocStorage.getInstance(
+            storageDirectory: Directory.current,
+          );
+          expect(
+            hydratedStorage.read('CounterBloc'),
+            isNull,
+          );
+        });
+
+        test('returns correct value when file exists', () async {
+          final file = File('./.hydrated_bloc.json');
+          file.writeAsStringSync(json.encode({
+            "CounterBloc": {"value": 4}
+          }));
+          hydratedStorage = await HydratedBlocStorage.getInstance(
+            storageDirectory: Directory.current,
+          );
+          expect(hydratedStorage.read('CounterBloc')['value'] as int, 4);
+        });
+
+        test(
+            'returns null value'
+            'when file exists but contains corrupt json and deletes the file',
+            () async {
+          final file = File('./.hydrated_bloc.json');
+          file.writeAsStringSync("invalid-json");
+          hydratedStorage = await HydratedBlocStorage.getInstance(
+            storageDirectory: Directory.current,
+          );
+          expect(hydratedStorage.read('CounterBloc'), isNull);
+          expect(file.existsSync(), false);
+        });
+      });
+
+      group('write', () {
+        test('writes to file', () async {
+          hydratedStorage = await HydratedBlocStorage.getInstance(
+            storageDirectory: Directory.current,
+          );
+          await hydratedStorage.write('CounterBloc', json.encode({"value": 4}));
+
+          expect(hydratedStorage.read('CounterBloc'), '{"value":4}');
+        });
+      });
+
+      group('heavy write', () {
+        test('writes heavily to file', () async {
+          final token = 'CounterBloc';
+          final directory = Directory.current;
+          hydratedStorage = await HydratedBlocStorage.getInstance(
+            storageDirectory: directory,
+          );
+          final tasks = Iterable.generate(120, (i) => i).map((i) async {
+            final record = Iterable.generate(
+              i,
+              (i) => Iterable.generate(i, (j) => 'Point($i,$j);').toList(),
+            ).toList();
+            final jsoned = json.encode(record);
+            hydratedStorage.write(token, jsoned); // no await here
+
+            dynamic written;
```

Comment:
```suggestion
            Map<String, dynamic> written;
```

Reply:
`HydratedStorage` supports writing `dynamic` states, and here we write `List<List<String>>`